### PR TITLE
Fixes integration with wslay to prevent accepting packets when in_buffer full

### PR DESCRIPTION
**Pull request by RadenTheFolf**
Added a fix to https://github.com/Redot-Engine/redot-engine/issues/31

Allows wslay to reject packets when in_buffer is nearly full. Doing so restores TCP's behavior to handle dropping and retransmission of packets instead of accepting all packets and dropping them in the application layer.

Issue stemmed from the implementation around Wslay not interrupting the reciving of packets when the in_buffer was nearly full leading to the loss of data.

### CODE REVIEW

File: modules/websocket/packet_buffer.h [hunk 1]
```diff
@@ -5,6 +5,8 @@
 /*                             GODOT ENGINE                               */
 /*                        https://godotengine.org                         */
 /**************************************************************************/
+/* Copyright (c) 2024-present Redot Engine contributors                   */
+/*                                          (see REDOT_AUTHORS.md)        */
```
Comment (RaidyQK) [inline modules/websocket/packet_buffer.h]: Unnecesary

**Review by RaidyQK — CHANGES_REQUESTED** [review]
Convert the 1 liners to 1 line logic, no need for extra brackets

**Comment by wesamdev** [comment]
looks good

**Comment by Spartan322** [comment]
This issue probably should be targeting the master branch, the commits should be squashed into a single one, the commit message and PR title should be descriptive of what this is fixing, and the description should say `Fixes #31`.

File: modules/websocket/packet_buffer.h (hunk 1 above)
Comment (Salvakiya) [reply modules/websocket/packet_buffer.h]: The Redot stance as far as I can tell is to change every copyright header to include those two lines. I think that is why its in here. But I have not been involved.

**Comment by Spartan322** [comment]
Closing in preference to #697.

**Pull request closed without merging**